diff -up slmodem-2.9.11-20080817/modem/modem_cmdline.c.daemon slmodem-2.9.11-20080817/modem/modem_cmdline.c
--- slmodem-2.9.11-20080817/modem/modem_cmdline.c.daemon	2008-03-29 13:04:16.000000000 +0100
+++ slmodem-2.9.11-20080817/modem/modem_cmdline.c	2008-10-15 23:45:38.000000000 +0200
@@ -74,6 +74,7 @@ unsigned int need_realtime = 1;
 unsigned int ring_detector = 0;
 #endif
 unsigned int use_alsa = 0;
+unsigned int daemonize = 0;
 unsigned int use_short_buffer = 0;
 const char *modem_group = "uucp";
 mode_t modem_perm  = 0660;
@@ -86,6 +87,7 @@ enum {
 	OPT_COUNTRY,
 	OPT_COUNTRYLIST,
 	OPT_ALSA,
+	OPT_DAEMON,
 	OPT_GROUP,
 	OPT_PERM,
 #ifdef MODEM_CONFIG_RING_DETECTOR
@@ -113,6 +115,7 @@ static struct opt {
 	{'c',"country","default modem country name",MANDATORY,STRING,"USA"},
 	{ 0 ,"countrylist","show list of supported countries"},
 	{'a',"alsa","ALSA mode (see README for howto)"},
+	{'d',"daemon","Daemon mode"},
 	{'g',"group","Modem TTY group",MANDATORY,STRING,"uucp"},
 	{'p',"perm","Modem TTY permission",MANDATORY,INTEGER,"0660"},
 #ifdef MODEM_CONFIG_RING_DETECTOR
@@ -252,6 +255,8 @@ void modem_cmdline(int argc, char *argv[
 #endif
 		use_alsa = 1;
 	}
+	if(opt_list[OPT_DAEMON].found)
+		daemonize = 1;
 	if(opt_list[OPT_GROUP].found)
 		modem_group = opt_list[OPT_GROUP].arg_val;
 	if(opt_list[OPT_PERM].found) {
diff -up slmodem-2.9.11-20080817/modem/modem_main.c.daemon slmodem-2.9.11-20080817/modem/modem_main.c
--- slmodem-2.9.11-20080817/modem/modem_main.c.daemon	2008-08-04 03:40:13.000000000 +0200
+++ slmodem-2.9.11-20080817/modem/modem_main.c	2008-10-15 23:49:00.000000000 +0200
@@ -57,6 +57,7 @@
 #include <signal.h>
 #include <limits.h>
 #include <grp.h>
+#include <syslog.h>
 
 #ifdef SUPPORT_ALSA
 #define ALSA_PCM_NEW_HW_PARAMS_API 1
@@ -70,8 +71,8 @@
 #include <modem.h>
 #include <modem_debug.h>
 
-#define INFO(fmt,args...) fprintf(stderr, fmt , ##args );
-#define ERR(fmt,args...) fprintf(stderr, "error: " fmt , ##args );
+#define INFO(fmt,args...) (daemonize) ? syslog(LOG_INFO, fmt, ##args) : fprintf(stderr, fmt , ##args );
+#define ERR(fmt,args...) (daemonize) ? syslog(LOG_ERR, "error: " fmt, ##args) : fprintf(stderr, "error: " fmt , ##args );
 
 #define DBG(fmt,args...) dprintf("main: " fmt, ##args)
 
@@ -131,6 +132,7 @@ static char outbuf[4096];
 #define INTERNAL_DELAY 40 /* internal device tx/rx delay: should be selfdetectible */
 
 extern unsigned use_alsa;
+extern unsigned daemonize;
 static snd_output_t *dbg_out = NULL;
 
 static int alsa_mixer_setup(struct device_struct *dev, const char *dev_name)
@@ -966,6 +968,11 @@ int modem_main(const char *dev_name)
 	INFO("modem `%s' created. TTY is `%s'\n",
 	     m->name, m->pty_name);
 
+	// daemonize
+	if (daemonize)
+		if (!daemon(0, 0))
+			ERR("slmodemd could not daemonize");
+
 	sprintf(path_name,"/var/lib/slmodem/data.%s",basename(dev_name));
 	datafile_load_info(path_name,&m->dsp_info);
 
@@ -984,7 +991,7 @@ int modem_main(const char *dev_name)
 	signal(SIGINT, mark_termination);
 	signal(SIGTERM, mark_termination);
 
-	INFO("Use `%s' as modem device, Ctrl+C for termination.\n",
+	INFO("Modem device: `%s'\n",
 	     *link_name ? link_name : m->pty_name);
 
 	/* main loop here */
